core-primitives/types/src/parentchain.rs

```diff
@@ -0,0 +1,42 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+//! Parentchain specific params. Be sure to change them if your node uses different types.
```

Comment:
Nice, did you move them from somewhere, or did you just plainly create them?

Reply:
I plainly created them. There were many hardcoded types (such as in core-primitives/node-api/api-client-extensions/src/chain.rs & core-primitives/node-api/api-client-extensions/src/account.rs)
or types used from the itp_types crate directly. Which is quite misleading because itp_types contains a `pub use itp_sgx_runtime_primitives::types::*;`.. so effectively using sgx-runtime types (such as in core/parentchain/test/src/parentchain_block_builder.rs)